server/game/cards/02_SHD/leaders/FinnThisIsARescue.ts

```diff
@@ -0,0 +1,50 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { LeaderUnitCard } from '../../../core/card/LeaderUnitCard';
+import { RelativePlayer, WildcardCardType } from '../../../core/Constants';
+
+export default class FinnThisIsARescue extends LeaderUnitCard {
+    protected override getImplementationId () {
+        return {
+            id: '9596662994',
+            internalName: 'finn#this-is-a-rescue',
+        };
+    }
+
+    private buildFinnAbilityTargetEffect() {
+        const upgradedCard = context => ({target: context.target.parentCard});
+        return AbilityHelper.immediateEffects.sequential([
+            AbilityHelper.immediateEffects.giveShield(upgradedCard),
+            AbilityHelper.immediateEffects.defeat(),
+        ]);
```

Comment:
I had trouble resolving the parent card of an upgrade after it was defeated. As a result, I ended up with giveShield before defeat. Happy to change this to be more accurate if there's a way to do it that I missed.

Reply:
We definitely need to keep them in the right order to ensure game state stays correct in between. Not sure exactly how that could cause an issue but weird edge cases always come up.

That said, I think I can make a small tweak to `sequential()` so that we could do something like this:

```typescript
  immediateEffect: AbilityHelper.immediateEffects.sequential((context) => {
      const upgradedCard = context.target.parentCard;
      return [
          AbilityHelper.immediateEffects.defeat(),
          AbilityHelper.immediateEffects.giveShield(upgradedCard),
      ];
  }) });
```

I'll do the patch for that later today.

Reply:
Okay I did a fix and it passes your current tests. Probably easier for you to just merge the branch into yours, could you merge `ammayberry1/finn-bug-fix` (branch link: https://github.com/SWU-Karabast/forceteki/tree/ammayberry1/finn-bug-fix)

Reply:
Ah okay so I did all of the above before really reviewing the code and realized that this is not the right implementation of his ability. Any time a card says "if you do" we need to use the ifYouDo property because there's some nuance around effect resolution that it handles correctly.

The `sequential` operation will do both the defeat and the giveShield operation nomatter what, but if something prevented the defeat from resolving then giveShield should also not resolve. Again, I'm not 100% if there's a case where that would come up currently, but best to use the right tool for the job. Take a look at Enforced Loyalty or Hondo Ohnaka for examples of how to use the `ifYouDo` property.